plugin-server/src/main/ingestion-queues/kafka-queue.ts

```diff
@@ -51,7 +59,64 @@ export class KafkaQueue implements Queue {
         }
     }
 
-    private async eachBatch({
+    private async eachMessageBuffer(
+        message: KafkaMessage,
+        resolveOffset: EachBatchPayload['resolveOffset']
+    ): Promise<void> {
+        const bufferEvent = JSON.parse(message.value!.toString())
+        await runInstrumentedFunction({
+            server: this.pluginsServer,
+            event: bufferEvent,
+            func: (_) => this.workerMethods.ingestBufferEvent(bufferEvent),
+            statsKey: `kafka_queue.ingest_buffer_event`,
+            timeoutMessage: 'After 30 seconds still running ingestBufferEvent',
+        })
+        resolveOffset(message.offset)
```

Comment:
@tiina303 re your comment about offsets - did you notice this?

From the `kafkajs` docs:

> resolveOffset() is used to mark a message in the batch as processed. In case of errors, the consumer will automatically commit the resolved offsets.